src/test/unit/huffman_unittest.cc

```diff
@@ -0,0 +1,466 @@
+    { 0xCA,  10, 0x0024 },  //  0000100100
+    { 0xCB,  10, 0x0023 },  //  0000100011
+    { 0xCC,  10, 0x0022 },  //  0000100010
+    { 0xCD,  10, 0x0021 },  //  0000100001
+    { 0xCE,  10, 0x0020 },  //  0000100000
+    { 0xCF,  10, 0x001F },  //  0000011111
+    { 0xD0,  10, 0x001E },  //  0000011110
+    { 0xD1,  10, 0x001D },  //  0000011101
+    { 0xD2,  10, 0x001C },  //  0000011100
+    { 0xD3,  10, 0x001B },  //  0000011011
+    { 0xD4,  10, 0x001A },  //  0000011010
+    { 0xD6,  10, 0x0019 },  //  0000011001
+    { 0xD7,  10, 0x0018 },  //  0000011000
+    { 0xD8,  10, 0x0017 },  //  0000010111
+    { 0xD9,  10, 0x0016 },  //  0000010110
+    { 0xDA,  10, 0x0015 },  //  0000010101
+    { 0xDB,  10, 0x0014 },  //  0000010100
+    { 0xDC,  10, 0x0013 },  //  0000010011
+    { 0xDE,  10, 0x0012 },  //  0000010010
+    { 0xDF,  10, 0x0011 },  //  0000010001
+    { 0xE1,  10, 0x0010 },  //  0000010000
+    { 0xE2,  10, 0x000F },  //  0000001111
+    { 0xE4,  10, 0x000E },  //  0000001110
+    { 0xEF,  10, 0x000D },  //  0000001101
+    { 0xD5,  11, 0x0019 },  //  00000011001
+    { 0xDD,  11, 0x0018 },  //  00000011000
+    { 0xE3,  11, 0x0017 },  //  00000010111
+    { 0xE5,  11, 0x0016 },  //  00000010110
+    { 0xE6,  11, 0x0015 },  //  00000010101
+    { 0xE7,  11, 0x0014 },  //  00000010100
+    { 0xE8,  11, 0x0013 },  //  00000010011
+    { 0xE9,  11, 0x0012 },  //  00000010010
+    { 0xEA,  11, 0x0011 },  //  00000010001
+    { 0xEB,  11, 0x0010 },  //  00000010000
+    { 0xEC,  11, 0x000F },  //  00000001111
+    { 0xED,  11, 0x000E },  //  00000001110
+    { 0xEE,  11, 0x000D },  //  00000001101
+    { 0xF2,  11, 0x000C },  //  00000001100
+    { 0xF3,  11, 0x000B },  //  00000001011
+    { 0xF4,  11, 0x000A },  //  00000001010
+    { 0xF5,  11, 0x0009 },  //  00000001001
+    { 0xF6,  11, 0x0008 },  //  00000001000
+    { 0xF7,  11, 0x0007 },  //  00000000111
+    { 0xF8,  11, 0x0006 },  //  00000000110
+    { 0xFA,  11, 0x0005 },  //  00000000101
+    { 0xFB,  11, 0x0004 },  //  00000000100
+    { 0xFC,  11, 0x0003 },  //  00000000011
+    { 0xFD,  11, 0x0002 },  //  00000000010
+    { 0xFE,  11, 0x0001 },  //  00000000001
+    { 0xF9,  12, 0x0001 },  //  000000000001
+    { HUFFMAN_EOF,  12, 0x0000 },  //  000000000000
+};
+
+int huffManLenIndex[HUFFMAN_TREE_SIZE];
+
+void huffmanInitDecodeLenIndex(void)
+{
+    // create an index of first code at each possible length
+    for (int ii = 0; ii < HUFFMAN_TREE_SIZE; ++ii) {
+        huffManLenIndex[ii] = -1;
+    }
+    for (int ii = 0; ii < HUFFMAN_TREE_SIZE; ++ii) {
+        if (huffManLenIndex[huffmanTree[ii].codeLen] == -1) {
+            huffManLenIndex[huffmanTree[ii].codeLen] = ii;
+        }
+    }
+}
+
+int huffmanDecodeBuf(uint8_t *outBuf, int outBufLen, const uint8_t *inBuf, int inBufLen, int inBufCharacterCount, const huffmanTree_t *huffmanTree)
+{
+    static bool initialized = false;
+    if (!initialized) {
+        huffmanInitDecodeLenIndex();
```

Comment:
It's a bit inconsistent that this function can be supplied with a custom `huffmanTree` but `huffmanInitDecodeLenIndex` assumes the default one.